### ISSUE TITLE
Can't find extension methods

### ISSUE THREAD

**Issue by Bartolomeus-649** [dotnet/core#2550]
# Unable to use extension method

### This does not work
```
Object data = new Object();
System.Net.Http.HttpClient client = new System.Net.Http.HttpClient();
bool result = await client.PostJsonAsync<bool>("api/TestController/TestMethod", data);
```
     
    
### But this works
```
Object data = new Object();
System.Net.Http.HttpClient client = new System.Net.Http.HttpClient();
bool result = await Microsoft.AspNetCore.Components.HttpClientJsonExtensions.PostJsonAsync<bool>(client, "api/TestController/TestMethod", data);

```

### Build error
`error CS1061: 'HttpClient' does not contain a definition for 'PostJsonAsync' and no accessible extension method 'PostJsonAsync' accepting a first argument of type 'HttpClient' could be found (are you missing a using directive or an assembly reference?)`

### The project file
```
<Project Sdk="Microsoft.NET.Sdk.Web">

  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
    <OutputType>Exe</OutputType>
    <RestoreAdditionalProjectSources>
      https://dotnet.myget.org/F/aspnetcore-dev/api/v3/index.json;
      https://dotnet.myget.org/F/blazor-dev/api/v3/index.json;
    </RestoreAdditionalProjectSources>
    <LangVersion>7.3</LangVersion>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.AspNetCore.Blazor" Version="0.9.0-preview3-19154-02" />
    <PackageReference Include="Microsoft.AspNetCore.Blazor.Build" Version="0.9.0-preview3-19154-02" PrivateAssets="all" />
    <PackageReference Include="Microsoft.AspNetCore.Components" Version="3.0.0-preview3-19153-02" />
    <PackageReference Include="Microsoft.NETCore.Platforms" Version="3.0.0-preview3.19128.7" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\BlazorTest4.Shared\BlazorTest4.Shared.csproj" />
  </ItemGroup>

</Project>

```
## General
- The code that work and the one that does not work are in the same method in the same file, the only thing I do is to comment out one of them at the time.
- The code is in a static method in a static class if that should matter for any reason. Does not work in a non static method in a non static class either .
- The error is present in design time as well, before building.
- Tried to directly reference the assembly in the project, no  change.

**Comment by NextTurn** [comment]
> Are you missing a using directive?

`using Microsoft.AspNetCore.Components;`

**Comment by Bartolomeus-649** [comment]
yes, I have that using.
But, it started working now after I updated VS 2019 to v 16.1.0 preview 1.0.
Also in this new version there is a suggestion to add the using, which I didn't get in the previous version.

So consider this resolved as of VS 2019 to v 16.1.0 preview 1.0.
